File: HereForYou/Entities/Site.cs
using System;
using System.Collections.Generic;
using LinqToDB.Mapping;

namespace HereForYou.Entities
{
    [Table("Site", IsColumnAttributeRequired = false)]
    public class Site
    {
        [Identity, PrimaryKey]
        public int Id { get; set; }

        public string Name { get; set; }
        public DateTime? LastVisit { get; set; }
        public string Address { get; set; }
    }

    [Table("Site", IsColumnAttributeRequired = false)]
    public class SiteExtended : Site
    {
        public List<SiteVisit> Visits { get; set; }
    }

    [Table("Site", IsColumnAttributeRequired = false)]
    public class SiteAgg : Site
    {
        public string WorkerNames { get; set; }
        public string TeamMembers { get; set; }
    }
}

File: Frontend/src/app/sites/site.ts
import { SiteVisit } from './site-visit';

export class Site {

  public name?: string;
  public lastVisit?: Date;
  public id?: number;
  public address?: string;

  constructor() {
  }
}

export class SiteExtended extends Site {
  public visits: SiteVisit[] = [];
}

export class SiteAgg extends Site {
  public workerNames: string;
  public teamMembers: string;
}
